flow/activities/flowable.go

```diff
@@ -346,7 +346,8 @@ func (a *FlowableActivity) SyncFlow(
 				slog.Int64("totalRecordsSynced", totalRecordsSynced.Load()))
 
 			if options.NumberOfSyncs > 0 && currentSyncFlowNum.Load() >= options.NumberOfSyncs {
-				break
+				close(syncDone)
+				return nil
```

Comment:
don't like this fix since returning early with parallel normalize can cause context cancellation, preventing normalization